### PR TITLE
SAMZA-2291: Custom IME Put in InMemoryManager runs into Offset mismatch even if offsets are correctly set

### PR DESCRIPTION
**Pull request by Sanil15**
- Fixed the untested code path
- Fixed the bug in InMemoryManager

### CODE REVIEW

**Comment by Sanil15** [comment]
@bharathkk @shanthoosh for review

File: samza-core/src/main/java/org/apache/samza/system/inmemory/InMemoryManager.java [hunk 1]
```diff
@@ -88,7 +88,7 @@ void put(SystemStreamPartition ssp, Object key, Object message) {
   void put(SystemStreamPartition ssp, IncomingMessageEnvelope envelope) {
     List<IncomingMessageEnvelope> messages = bufferedMessages.get(ssp);
     String offset = String.valueOf(messages.size());
-    if (envelope.getOffset().equals(offset)) {
+    if (!envelope.getOffset().equals(offset)) {
```
Comment (shanthoosh) [inline samza-core/src/main/java/org/apache/samza/system/inmemory/InMemoryManager.java]: Would be great to check if the envelope.getOffset() is not null.

Comment (Sanil15) [reply samza-core/src/main/java/org/apache/samza/system/inmemory/InMemoryManager.java]: Done

**Comment by shanthoosh** [comment]
@Sanil15 
Merged the patch to trunk. Thanks for the bug-fix.

**Pull request merged**